Testnet/DAOContract/DAOContract/DAOContract.cs

```diff
@@ -0,0 +1,301 @@
+﻿using Stratis.SmartContracts;
+using System;
+
+[Deploy]
+public class DAOContract : SmartContract
+{
+    public Address Owner
+    {
+        get => State.GetAddress(nameof(Owner));
+        private set => State.SetAddress(nameof(Owner), value);
+    }
+
+    public uint MinQuorum => WhitelistedCount / 2 + 1;
+
+    public uint WhitelistedCount
+    {
+        get => State.GetUInt32(nameof(WhitelistedCount));
+        private set => State.SetUInt32(nameof(WhitelistedCount), value);
+    }
+
+    public uint MinVotingDuration
+    {
+        get => State.GetUInt32(nameof(MinVotingDuration));
+        private set => State.SetUInt32(nameof(MinVotingDuration), value);
+    }
+
+    public uint MaxVotingDuration
+    {
+        get => State.GetUInt32(nameof(MaxVotingDuration));
+        private set => State.SetUInt32(nameof(MaxVotingDuration), value);
+    }
+
+    public uint LastProposalId
+    {
+        get => State.GetUInt32(nameof(LastProposalId));
+        private set => State.SetUInt32(nameof(LastProposalId), value);
+    }
+
+    public bool IsWhitelisted(Address address) => State.GetBool($"Whitelisted:{address}");
+    private void SetIsWhitelisted(Address address, bool allowed) => State.SetBool($"Whitelisted:{address}", allowed);
+
+    public ulong GetVotingDeadline(uint proposalId) => State.GetUInt64($"Deadline:{proposalId}");
+    private void SetVotingDeadline(uint proposalId, ulong block) => State.SetUInt64($"Deadline:{proposalId}", block);
+
+    public uint GetYesVotes(uint proposalId) => State.GetUInt32($"YesVotes:{proposalId}");
+    private void SetYesVotes(uint proposalId, uint value) => State.SetUInt32($"YesVotes:{proposalId}", value);
+
+    public uint GetNoVotes(uint proposalId) => State.GetUInt32($"NoVotes:{proposalId}");
+    private void SetNoVotes(uint proposalId, uint value) => State.SetUInt32($"NoVotes:{proposalId}", value);
+
+    public uint GetVote(uint proposalId, Address address) => State.GetUInt32($"Vote:{proposalId}:{address}");
+    private void SetVote(uint proposalId, Address address, Votes vote) => State.SetUInt32($"Vote:{proposalId}:{address}", (uint)vote);
+
+    public Proposal GetProposal(uint index) => State.GetStruct<Proposal>($"Proposals:{index}");
+
+    private void SetProposal(uint index, Proposal proposal) => State.SetStruct($"Proposals:{index}", proposal);
+
+    public const uint DefaultMaxDuration = 60u * 60 * 24 * 7 / 16; // 1 week period as second/ block duration as second
+    public DAOContract(ISmartContractState state, uint minVotingDuration)
+        : base(state)
+    {
+        Assert(DefaultMaxDuration > minVotingDuration, $"MinVotingDuration should be lower than maxVotingDuration({DefaultMaxDuration})");
+
+        Owner = Message.Sender;
+        LastProposalId = 1;
+        MinVotingDuration = minVotingDuration;
+        MaxVotingDuration = DefaultMaxDuration;
+    }
+
+    public uint CreateProposal(Address recipent, ulong amount, uint votingDuration, string description)
+    {
```

Comment:
Prevent funds from being sent here.

```suggestion
        Assert(Message.Value == 0, "Must not include funds");
        
```